src/visualization/convert.jl

```diff
@@ -73,6 +73,53 @@ function get_data_2d(center_level_0, length_level_0, leaf_cells, coordinates, le
 end
 
 
+# Extract data from a 1D DG solution and prepare it for visualization as a line plot.
+# This returns a tuple with
+# - x coordinates
+# - nodal 1D data
+#
+# Note: This is a low-level function that is not considered as part of Trixi's interface and may
+#       thus be changed in future releases.
+function get_data_1d(original_nodes, unstructured_data, nvisnodes)
+  # Get the dimensions of u; where n_vars is the number of variables, n_nodes the number of nodal values per element and n_elements the total number of elements.
+  n_nodes, n_elements, n_vars = size(unstructured_data)
+
+  # Set the amount of nodes visualized according to nvisnodes.
+  if nvisnodes === nothing
+    max_nvisnodes = 2 * n_nodes
+  elseif nvisnodes == 0
+    max_nvisnodes = n_nodes
+  else
+    @assert nvisnodes >= 2 "nvisnodes must be zero or >= 2"
+    max_nvisnodes = nvisnodes
+  end
+
+  interpolated_nodes = Array{eltype(original_nodes),    2}(undef, max_nvisnodes, n_elements)
+  interpolated_data  = Array{eltype(unstructured_data), 3}(undef, max_nvisnodes, n_elements, n_vars)
+
+  for j in 1:n_elements
+    # Interpolate on an equidistant grid.
+    interpolated_nodes[:, j] .= range(original_nodes[1,1,j], original_nodes[1,end,j], length = max_nvisnodes)
+  end
+
+  nodes_in, _ = gauss_lobatto_nodes_weights(n_nodes)
+  nodes_out = collect(range(-1, 1, length = max_nvisnodes))
+
+  # Calculate vandermonde matrix for interpolation.
+  vandermonde = polynomial_interpolation_matrix(nodes_in, nodes_out)
+
+  # Iterate over all variables.
+  for v in 1:n_vars
+    reshaped_data = reshape(unstructured_data[:, :, v], 1, n_nodes, n_elements)
+    # Interpolate data for each element.
+    for element in 1:n_elements
+      interpolated_data[:, element, v] = vec(multiply_dimensionwise(vandermonde, reshaped_data[:, :, element]))
+    end
+  end
+  # Return results after data is reshaped
+  return vec(interpolated_nodes), collect(reshape(interpolated_data, :, n_vars))
```

Comment:
```suggestion
  return vec(interpolated_nodes), reshape(interpolated_data, :, n_vars)
```
Why should we allocate again?